ext/neural_snow/ModelTools.jl

```diff
@@ -428,4 +785,121 @@ function trainmodel!(
+
+A helper/utility function for writing the weights of the PREDICTIVE COMPONENT
+of a model made with `make_model()` or `make_model_paper()` to a human-parseable text file.
+
+This only saves the weights of the predictive component of the network, and only
+if the predictive component is a chain of named `Dense` layers, in the same manner
+as the paper by Charbonneau et. al. (2025). For custom models or alternative constructions
+we refer the reader to the Flux standard Saving and Loading page: https://fluxml.ai/Flux.jl/stable/guide/saving/
+This provides parameter files that can be readily used with other coding languages and is human readable,
+as well as working independent of the Flux package version.
+
+# Arguments
+- `path`::String: The filepath for which to save the text file.
+- `model`: the model for which to save the predictive parameters.
+"""
+function save_predictive_model_weights(path::String, model::Chain)
+    layers = model[:pred].layers.layers
+    scalings = layers.scale.in_scales
+    open(path, "w") do file
+        write(file, "SCALING:\n")
+        println(file, join(scalings, " "))
+        for layer in collect(keys(layers))[2:end]
+            write(file, "\n" * string(layer) * "-WEIGHT:\n")
+            for row in eachrow(layers[layer].weight)
+                println(file, join(row, " "))
+            end
+            write(file, "\n" * string(layer) * "-BIAS:\n")
+            println(file, join(layers[layer].bias, " "))
+        end
+        write(file, "\nFINALSCALE:\n")
+        println(file, -model[:apply_relus].weight[end, 3])
+    end
+end
+
+
+"""
+   load_model_weights!(path, model)
+
+A helper/utility function for loading the weights of the PREDICTIVE COMPONENT
+into a model made with `make_model()` or `make_model_paper()` from a text file (or web URL) for a model
+with the same structure made with `save_predictive_model_weights()`.
+
+Note this will only work for models matching the structure as those defined
+in the text file, and is written specifically for models made with models
+generated with `make_model()` or `make_model_paper()`. Calling this function will not work for custom
+model types, or for models having different structures than those which had their
+parameters saved with `save_predictive_model_weights()`. For custom models or alternative constructions
+we refer the reader to the Flux standard Saving and Loading page: https://fluxml.ai/Flux.jl/stable/guide/saving/
+This provides a loading utility that can be readily used with weight files generated from other coding languages,
+and in a way that is stable to changing versions of the Flux package.
+
+# Arguments
+- `path`::String: The filepath (or URL) from where to get the text file.
+- `model`: the model for which to load in the predictive parameters.
+- `FT` : Optional arg for passing the right Float type if not inferred from passed model.
+"""
+function load_model_weights!(filepath::String, model::Chain; FT = Float32)
```

Comment:
We need a test to make sure that reading and writing are compatible operations (ie, a model read is identical to the model that wrote the corresponding file)